makepki.sh

```diff
@@ -14,124 +14,46 @@
 # ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 # OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
-domain=example.com
-server1=server1.example.com
-server2=server2.example.com
-client=client.example.com
+dir=pki
 dbdir=nssdb
 password=password
 
-profile_ca_request_options=(-1 -2 -4)
-profile_ca_request_input="\$'0\n1\n5\n6\n9\ny\ny\n\ny\n1\n7\nfile://'\$(readlink -f \$dbdir)/\$ca.crl\$'\n-1\n-1\n-1\nn\nn\n'"
-profile_ca_create_options=(-v 120)
-profile_ca_add_options=(-t CT,C,C)
-profile_server_request_options=(-4)
-profile_server_request_input="\$'1\n7\nfile://'\$(readlink -f \$dbdir)/\$ca.crl\$'\n-1\n-1\n-1\nn\nn\n'"
-profile_server_create_options=(-v 12)
-profile_server_add_options=(-t ,,)
+if [ ! -d "$dir" ]; then
+    scriptdir=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+    "$scriptdir"/makepki.py
+fi
 
-gen_cert() {
-    local profile="$1" nick="$2" subject="$3" ca request_options request_input create_options serial add_options pwfile noise csr crt
-    shift 3
+pwfile="$(mktemp)"
+echo "$password" >"$pwfile"
 
-    ca="$(dirname $nick)"
-    if [ "$ca" = "." ]; then
-        ca="$nick"
-    fi
-
-    eval "request_options=(\"\${profile_${profile}_request_options[@]}\")"
-    eval "eval request_input=\"\${profile_${profile}_request_input}\""
-
-    eval "create_options=(\"\${profile_${profile}_create_options[@]}\")"
-    if [ "$ca" = "$nick" ]; then
-        create_options=("${create_options[@]}" -x -m 1)
-    else
-        eval "serial_${ca//\//_}=\$((\${serial_${ca//\//_}:-1}+1))"
-        eval "serial=\$serial_${ca//\//_}"
-        create_options=("${create_options[@]}" -c "$ca" -m "$serial")
-    fi
+if [ ! -d "$dbdir" ]; then
+    mkdir "$dbdir"
+    certutil -N -d "$dbdir" -f "$pwfile"
+fi
 
-    eval "add_options=(\"\${profile_${profile}_add_options[@]}\")"
-
-    pwfile="$(mktemp)"
-    echo "$password" >"$pwfile"
-
-    noise="$(mktemp)"
-    head -c 20 /dev/urandom >"$noise"
-
-    if [ ! -d "$dbdir" ]; then
-        mkdir "$dbdir"
-        certutil -N -d "$dbdir" -f "$pwfile"
-    fi
+find "$dir" -name \*.crt | while read certfile; do
```

Comment:
Would it be feasible to use the Python OpenSSL bindings, create the pkcs12 files using just that and drop the NSS databases cruft? That way we don't have to split the logic into two scripts written in two different languages.
Do we need the NSS database for something, anyway?